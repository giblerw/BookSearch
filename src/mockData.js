export const emptySet = {
    currentQuery: 'default',
    listData: {},
    totalItems: 0,
    activeSort: 'relevance',
}

export const set1 = {
    items: [{
        volumeInfo: {
            title: 'Testament to Youth',
            authors: ['Vera Brittain'],
            description: `This classic memoir of the First World War is now a major motion picture starring Alicia Vikander and Kit Harington. Includes an afterword by Kate Mosse OBE. In 1914 Vera Brittain was 20, and as war was declared she was preparing to study at Oxford. Four years later her life - and the life of her whole generation - had changed in a way that would have been unimaginable in the tranquil pre-war era. TESTAMENT OF YOUTH, one of the most famous autobiographies of the First World War, is Brittain's account of how she survived those agonising years; how she lost the man she loved; how she nursed the wounded and how she emerged into an altered world. A passionate record of a lost generation, it made Vera Brittain one of the best-loved writers of her time, and has lost none of its power to shock, move and enthral readers since its first publication in 1933.`,
            imageLinks: {
                smallThumbnail: 'http://books.google.com/books/content?id=YSI4AgAAQBAJ&printsec=frontcover&img=1&zoom=5&edge=curl&source=gbs_api'
            }
        }
    }],
    totalItems: 1681
}

export const set2 = {
    items: [{
        volumeInfo: {
            title: 'Hoot',
            authors: ['Carl Hiaasen'],
            description: `Roy, who is new to his small Florida community, befriends a mysterious boy who introduces him to potty-trained alligators, burrowing owls, and sparkly-tailed poisonous snakes. A Newbery Honor Book, ALA Notable Book, & ALA Best Book for Young Adults. Reissue.`,
            imageLinks: {
                smallThumbnail: 'http://books.google.com/books/content?id=BTDSfx5bunwC&printsec=frontcover&img=1&zoom=5&edge=curl&source=gbs_api'
            }
        }
    }],
    totalItems: 1681
};

export const set3 = {
    items: [{
        volumeInfo: {
            title: 'Testament to Youth',
            authors: null,
            description: '',
            imageLinks: {
                smallThumbnail: 'http://books.google.com/books/content?id=YSI4AgAAQBAJ&printsec=frontcover&img=1&zoom=5&edge=curl&source=gbs_api'
            }
        }
    }],
    totalItems: 1681
}

export const beforeSearch = {
    query: '',
    searched: false
}

export const afterSearch = {
    query: 'harry',
    searched: true
}