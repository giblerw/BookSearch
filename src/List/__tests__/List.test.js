import { shallowMount, mount } from '@vue/test-utils';
import ListComponent from '../index.vue';
import * as mockData from '../../mockData';

describe('List.vue', () => {
    it('renders the expected HTML structure, without data', () => {
        const wrapper = mount(ListComponent, {
            setData: mockData.emptySet
        });
        
        expect(wrapper.element).toMatchSnapshot();
    });

    it('renders the expected HTML structure, with data', () => {
        const wrapper = mount(ListComponent, {
            setData: mockData.set1
        });
        
        expect(wrapper.element).toMatchSnapshot();
    });

    it('should render prompt before search', () => {
        const wrapper = shallowMount(ListComponent);
        
        expect(wrapper.find('[class="list-prompt"]').text())
        .toEqual('Submit a search above to get Book data.');
        
        expect(wrapper.element).toMatchSnapshot();
    });

    it('should render results prompt after search', () => {
        const wrapper = shallowMount(ListComponent);
        wrapper.setData({
            currentQuery: 'harry potter',
            listData: {
                'harry potter': {
                    items: [],
                    kind: 'books#volumes',
                    totalItems: 2770
                }
            },
            totalItems: 2770,
        })

        expect(wrapper.find('[class="list-prompt"]').text())
        .toEqual('Search found 2770 results for: harry potter');

        expect(wrapper.element).toMatchSnapshot()
    });
});