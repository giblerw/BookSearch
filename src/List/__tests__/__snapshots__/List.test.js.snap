// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List.vue renders the expected HTML structure, with data 1`] = `
<div
  class="list-prompt"
>
  <p
    id="search-propmt"
  >
    
    Submit a search above to get Book data.
  
  </p>
   
  <!---->
</div>
`;

exports[`List.vue renders the expected HTML structure, without data 1`] = `
<div
  class="list-prompt"
>
  <p
    id="search-propmt"
  >
    
    Submit a search above to get Book data.
  
  </p>
   
  <!---->
</div>
`;

exports[`List.vue should render prompt before search 1`] = `
<div
  class="list-prompt"
>
  <p
    id="search-propmt"
  >
    
    Submit a search above to get Book data.
  
  </p>
   
  <!---->
</div>
`;

exports[`List.vue should render results prompt after search 1`] = `
<div
  class="list-prompt"
>
  <!---->
   
  <div>
    <h6
      id="result-propmt"
    >
      Search found 
      <b>
        2770
      </b>
       results for: 
      <em>
        <b
          id="query-read"
        >
          harry potter
        </b>
      </em>
    </h6>
     
    <!---->
  </div>
</div>
`;
